cmd/oras/root/manifest/index/update_test.go

```diff
@@ -16,15 +16,96 @@ limitations under the License.
 package index
 
 import (
-	"os"
+	"context"
+	"fmt"
 	"reflect"
 	"testing"
 
 	"github.com/opencontainers/go-digest"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
-	"oras.land/oras/cmd/oras/internal/output"
+	"oras.land/oras-go/v2"
+	"oras.land/oras/cmd/oras/internal/display/status"
 )
 
+// // for quick drafting
+// type badWriter struct {
+// }
+
+// func (bw *badWriter) Write(p []byte) (n int, err error) {
+// 	return 0, fmt.Errorf("test error")
+// }
+
+type testUpdateDisplayStatus struct {
+	onFetchingError        bool
+	onFetchedError         bool
+	onIndexPackedError     bool
+	onIndexPushedError     bool
+	onManifestRemovedError bool
+	onManifestAddedError   bool
+	onIndexMergedError     bool
+}
+
+func (tds *testUpdateDisplayStatus) OnFetching(manifestRef string) error {
+	if tds.onFetchingError {
+		return fmt.Errorf("OnFetching error")
+	}
+	return nil
+}
+
+func (tds *testUpdateDisplayStatus) OnFetched(manifestRef string, desc ocispec.Descriptor) error {
+	if tds.onFetchedError {
+		return fmt.Errorf("OnFetched error")
+	}
+	return nil
+}
+
+func (tds *testUpdateDisplayStatus) OnIndexPacked(desc ocispec.Descriptor) error {
+	if tds.onIndexPackedError {
+		return fmt.Errorf("error")
+	}
+	return nil
+}
+
+func (tds *testUpdateDisplayStatus) OnIndexPushed(path string) error {
+	if tds.onIndexPushedError {
+		return fmt.Errorf("error")
+	}
+	return nil
+}
+
+func (tds *testUpdateDisplayStatus) OnManifestRemoved(digest digest.Digest) error {
+	if tds.onManifestRemovedError {
+		return fmt.Errorf("OnManifestRemoved error")
+	}
+	return nil
+}
+
+func (tds *testUpdateDisplayStatus) OnManifestAdded(manifestRef string, desc ocispec.Descriptor) error {
+	if tds.onManifestAddedError {
+		return fmt.Errorf("error")
+	}
+	return nil
+}
+
+func (tds *testUpdateDisplayStatus) OnIndexMerged(indexRef string, desc ocispec.Descriptor) error {
+	if tds.onIndexMergedError {
+		return fmt.Errorf("error")
+	}
+	return nil
+}
+
+func NewTestUpdateDisplayStatus(onFetching, onFetched, onIndexPacked, onIndexPushed, onManifestRemoved, onManifestAdded, onIndexMerged bool) status.ManifestIndexUpdateHandler {
```

Comment:
This function is hard to maintain since it is not friendly to future changes. E.g. whenever we need to add a new handler test, we need to modify ALL the existing usage of this function in ALL the test cases.

How about assembling the structure directly in the case array?

E.g.

For example:
```diff
{
    name:          "remove one matched item",
    manifests:     []ocispec.Descriptor{A, B, C},
    digestSet:     map[digest.Digest]bool{B.Digest: false},
-	displayStatus: NewTestUpdateDisplayStatus(false, false, false, false, false, false, false),
+	displayStatus: &testUpdateDisplayStatus{},
    indexRef:      "test01",
    want:          []ocispec.Descriptor{A, C},
    wantErr:       false,
}
```

Or

```diff
{
    name:          "handler error",
    manifests:     []ocispec.Descriptor{A, B, C},
    digestSet:     map[digest.Digest]bool{B.Digest: false},
-	displayStatus: NewTestUpdateDisplayStatus(false, false, false, false, true, false, false),
+	displayStatus: &testUpdateDisplayStatus{onManifestRemovedError: true},
    indexRef:      "test07",
    want:          nil,
    wantErr:       true,
},
```

Reply:
Of course you can further apply a decorator pattern like `NewTestUpdateStatus().WithManifestRemovedError().WithManifestAddedError()`. 
 
 Luckily, I don't see usage of `NewTestUpdateDisplayStatus(true, true, true, true, true, true, true)` for now it's over design.

Reply:
Thanks for the comment. Removed the NewTestDisplayStatus functions.